src/modules/map/components/openlayers/utils/useDragBoxSelect.composable.js

```diff
@@ -20,16 +28,63 @@ export function useDragBoxSelect() {
     )
     dragBoxSelect.on('boxend', () => {
         const selectExtent = dragBoxSelect.getGeometry()?.getExtent()
+        if (selectExtent?.length !== 4) return
 
-        if (selectExtent?.length === 4) {
-            store.dispatch('click', {
-                clickInfo: new ClickInfo({ coordinate: selectExtent }),
-                ...dispatcher,
+        const dragBox = polygon(dragBoxSelect.getGeometry()?.getCoordinates())
+        const visibleLayers = store.getters.visibleLayers.filter((layer) =>
+            [LayerTypes.GEOJSON, LayerTypes.GPX, LayerTypes.KML].includes(layer.type)
+        )
+        const features = visibleLayers
+            .flatMap((layer) => {
+                if (layer.type === LayerTypes.KML) {
+                    const kmlFeatures = parseKml(layer, store.state.position.projection, [])
+                    return kmlFeatures.map((feature) => ({ feature: feature, layer }))
+                }
+                if (layer.type === LayerTypes.GPX) {
+                    const gpxFeatures = parseGpx(layer.gpxData, store.state.position.projection, [])
+                    return gpxFeatures.map((feature) => ({ feature: feature, layer }))
+                }
+                if (layer.type === LayerTypes.GEOJSON) {
+                    const geojsonFormat = new GeoJSON()
+                    const olFeatures = geojsonFormat.readFeatures(layer.geoJsonData, {
+                        featureProjection: store.state.position.projection.epsg,
+                    })
+                    return olFeatures.map((feature) => ({ feature: feature, layer }))
+                }
             })
-        }
+            .filter((result) => {
+                const geometry = fromOlGeometryToTurfGeometry(result.feature.getGeometry())
+                return geometry && dragBox && booleanIntersects(dragBox, geometry)
+            })
+            .map(({ feature, layer }) => createLayerFeature(feature, layer))
```

Comment:
I'm not sure it is a good idea to put this here, it only select GeoJSON, GPX and KML features, where is the code to select other features types ? In my opinion both code should sit next to each other to improve code organization.

Reply:
the other features are selected in click-on-map-management.plugin.js
the features of the GeoJSON, GPX and KML files are added to the ClickInfo features which are then also used in the click-on-map-management.plugin.js so i think the code is well placed here, i can change the name of the features to vectorFeatures to make it clear that those are just the vector features